core/src/main/java/com/hcl/ecomm/core/services/impl/ProductServiceImpl.java

```diff
@@ -0,0 +1,166 @@
+package com.hcl.ecomm.core.services.impl;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.hcl.ecomm.core.config.ProductServiceConfig;
+import com.hcl.ecomm.core.services.LoginService;
+import com.hcl.ecomm.core.services.ProductService;
+import com.hcl.ecomm.core.utility.ProductUtility;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.metatype.annotations.Designate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Component(immediate = true, enabled = true, service = ProductService.class)
+@Designate(ocd = ProductServiceConfig.class)
+public class ProductServiceImpl implements ProductService {
+
+	private static final Logger LOG = LoggerFactory.getLogger(ProductServiceImpl.class);
+
+	@Reference
+	LoginService loginService;
+
+	String scheme = "http";
+	String responseStream = null;
+	JsonArray productJsonArray = null;
+
+	@Activate
+	private ProductServiceConfig config;
+
+	
+	@Override
+	public String getDomainName() {
+		return config.productService_domainName();
+	}
+
+	@Override
+	public String getServicePath() {
+		return config.productService_servicePath();
+	}
+
+	@Override
+	public String getSearchCriteriaField() {
+		return config.productService_searchCriteriaField();
+	}
+
+	@Override
+	public String getSearchCriteriaValue() {
+		return config.productService_searchCriteriaValue();
+	}
+
+	
+
+	@Override
+	public JsonArray getAllProductDetails() {
+
+		LOG.info("inside getAllProductDetails method");
```

Comment:
@nitheshkumar-hcl : pl remove un-necessary logs

Reply:
Committed the changes as per the review comments